src/objects/codeholder.rs

```diff
@@ -1,3 +1,27 @@
+use super::constant::Constant;
 use super::instruction::Instruction;
 
-pub struct CodeHolder(pub Vec<Instruction>);
\ No newline at end of file
+/// A CodeHolder represents a set of executable instructions and a pool of immutable data for an
+/// [`crate::Interpreter`] to use at runtime.
+pub struct CodeHolder {
+    /// A [`Vec`] of executable instructions
+    pub instructions: Vec<Instruction>,
+    /// A pool of immutable data that is available to the VM at runtime.
+    pub constant_pool: Vec<Constant>,
```

Comment:
I don't think `constant_pool` should be public as it's more of a implementation detail

Reply:
I don't know of any way to manipulate constants externally, so I made it public for now until there is an official way to manipulate constants.

Reply:
`pub(crate)` should work

Reply:
I would like to go ahead and merge it now and then modify it once a public-facing way to manipulate constants is added